File: mobile/lib/features/auth/domain/entidades/usuario.dart
// Entidad Usuario (espejo del backend, sin password)

class Usuario {
  final int id;
  final String nombres;
  final String apellidos;
  final String email;
  final String? whatsapp;
  final bool emailVerificado;

  const Usuario({
    required this.id,
    required this.nombres,
    required this.apellidos,
    required this.email,
    this.whatsapp,
    required this.emailVerificado,
  });

  String get nombreCompleto => '$nombres $apellidos';

  factory Usuario.fromJson(Map<String, dynamic> json) => Usuario(
        id: json['id'] as int,
        nombres: json['nombres'] as String,
        apellidos: json['apellidos'] as String,
        email: json['email'] as String,
        whatsapp: json['whatsapp'] as String?,
        emailVerificado: (json['email_verificado'] as bool?) ?? false,
      );

  Map<String, dynamic> toJson() => {
        'id': id,
        'nombres': nombres,
        'apellidos': apellidos,
        'email': email,
        'whatsapp': whatsapp,
        'email_verificado': emailVerificado,
      };
}

File: backend/src/domain/entidades/usuario.ts
// Entidad Usuario según el diagrama de clases
// (codigo_estudiante removido por D-04: pertenece a la inscripción, E3)

export interface Usuario {
  id: number;
  nombres: string;
  apellidos: string;
  email: string;
  password: string; // hash bcrypt
  whatsapp: string | null;
  rol_id: number;
  rol_nombre: 'admin' | 'docente_estudiante';
  activo: boolean;
  email_verificado: boolean;
  // SaaS por cuenta (17/07, rediseñado 17/08): plan_id es el plan
  // contratado actualmente (Gratis por defecto al registrarse, null solo
  // en cuentas admin). trial_inicio ELIMINADO (17/08): el plan Gratis es
  // permanente, no hay nada que cronometrar — ver ObtenerEstadoCuenta.
  plan_id: number | null;
}

export type UsuarioPublico = Omit<Usuario, 'password'>;

export function aPublico(u: Usuario): UsuarioPublico {
  const { password: _omitido, ...publico } = u;
  return publico;
}
